src/backends/github/implementation.js

```diff
@@ -137,8 +137,19 @@ export default class GitHub {
       const response = await this.api.persistFiles(null, [mediaFile], options);
       const repo = this.repo || this.getRepoFromResponseUrl(response.url);
       const { value, size, path, fileObj } = mediaFile;
-      const url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
-      return { id: response.sha, name: value, size: fileObj.size, url, path: trimStart(path, '/') };
+      let url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
+
+      // Assets uploaded to private repos will need valid access tokens.
+      const isPrivateRepo = await this.api.isPrivateRepo();
+      if (isPrivateRepo) {
+        const files = await this.api.listFiles(this.config.get('media_folder'));
+        const file = files.find(f => (f.sha === mediaFile.sha));
+        if (file) {
+          url = file.download_url;
+        }
+      }
+
+      return { id: mediaFile.sha, name: value, size: fileObj.size, url, path: trimStart(path, '/') };
     }
```

Comment:
It looks like we're returning the file SHA now, instead of the response (commit) SHA. Maybe I just missed the reason for this, can you explain? :smile:

Reply:
@tech4him1 `response.sha` is undefined (yep, it was setting an undefined id before), at most it would have been `response.object.sha`. I set it to the file's sha instead to be more consistent with the other ids from when the media library initializes: https://github.com/netlify/netlify-cms/blob/master/src/backends/github/implementation.js#L94-L104

It's not really part of the original issue, true. But I thought since I was already in that very same line of code, I'd throw that in. 😅

Reply:
@Quicksaver Sounds good, maybe it should be split into a seperate commit? (same PR is fine)

Reply:
@tech4him1 makes sense, I split it into a separate commit. Also, I just stumbled upon a redundant file.filter call in `getMedia()` in implementation.js (`listFiles()` in API.js already has an exact same one) so I went ahead and created a commit for it as well. Let me know if I'm filling this PR too much with non-relevant commits. 😁